consensus/polybft/polybft.go

```diff
@@ -361,6 +362,34 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
 	}
 }
 
+func ForkManagerFactory(forks *chain.Forks) error {
```

Comment:
Why having this in the consensus? look like this belongs to the root

Reply:
handlers can depend on consensus. Current code i have put there probably can be moved in root, but handler registration can't (look at the example pr)

Reply:
I have refactored this a little bit. Server do all the forks registration and activation but factory method register all the handlers. If there are no factory method then consensus does not uses fork manager at all

https://github.com/0xPolygon/polygon-edge/pull/1544/commits/37e021cfa5311fa38e4a120d16ad8fe9793e75be